File: Leet Code/Python/problem_125.py
import re

class Solution(object):
    def isPalindrome(self, s):
        """
        :type s: str
        :rtype: bool
        """
        if (s == " "):
            return True

        myStr = s
        # lowerStr = myStr.lower()
        cleanStr = re.sub('[^0-9a-zA-Z]+', '', myStr.lower())
        reverseStr = cleanStr[::-1]
        return cleanStr == reverseStr
        


s = Solution()
result = s.isPalindrome("A man, a plan, a canal: Panama")
print(result)

File: Leet Code/JavaScript/problem_125.js
const isPalindrome = function(s) {
    if (s === " ") {
        return true;
    };

    let cleanStr = s.replace(/[^A-Za-z0-9]/gi, '');
    let lowerStr = cleanStr.toLowerCase();
    let reverseStr = "";

    for (let i = lowerStr.length - 1; i >= 0; i--){
        reverseStr = reverseStr + lowerStr[i];
    };

    console.log(lowerStr);
    console.log(reverseStr);
    return lowerStr === reverseStr;
};

const result = isPalindrome("hello / wor,@ld");
console.log(result);